headers/types/dungeon_mode/dungeon.h

```diff
@@ -1613,23 +1627,9 @@ struct dungeon {
     uint16_t grab_bag_item_weights[1416];
     // 0x2BE42: Spawn weights for secret room items in treasure boxes.
     // Same format as regular_item_weights.
-    uint16_t secret_room_item_weights[1416];
-    undefined field_0x2c952;
-    undefined field_0x2c953;
-    undefined field_0x2c954;
-    undefined field_0x2c955;
-    undefined field_0x2c956;
-    undefined field_0x2c957;
-    undefined field_0x2c958;
-    undefined field_0x2c959;
-    undefined field_0x2c95a;
-    undefined field_0x2c95b;
-    undefined field_0x2c95c;
-    undefined field_0x2c95d;
-    undefined field_0x2c95e;
-    undefined field_0x2c95f;
-    undefined field_0x2c960;
-    undefined field_0x2c961;
+    uint16_t secret_room_item_weights[1400];
```

Comment:
Hmm, @End45 what do you think of this? The trap weights thing seems reasonable, but it also seems weird that `secret_room_item_weights` would be shorter than all the other lists before it.

Reply:
A strange situation indeed. The code pointed out by Hecka does access an address that is included in the last positions of the array and starts writing trap data there. I guess this change works, but it would be a good idea to point out the inconsistency in a comment.

Reply:
Upon further review, I'm pretty sure the trap weight array is 25 long and I've updated it to match (because I realized the loop starts at 0 and thus is 1 longer than I initially thought). Will leave a comment about the inconsistency around the secret room weights.